deps/build.jl

```diff
@@ -3,29 +3,77 @@ using BinaryProvider, SHA # requires BinaryProvider 0.3.0 or later
 using METIS4_jll
 using OpenBLAS32_jll
 
+struct HSLVersion
+    algname::String
+    version::String
+    sha::String
+    ext::String
+end
+
+getname(ver::HSLVersion) = ver.algname * "-" * ver.version * ver.ext 
+
+function checksha(version::HSLVersion, filepath)
+    if isfile(filepath)
+        open(filepath) do f
+            return bytes2hex(sha256(f)) == version.sha 
+        end
+    else
+        return false
+    end
+end
+
+function findversion(versions::Vector{HSLVersion}, path::AbstractString)
+    for ver in versions
+        archivename = getname(ver) 
+        archivepath = joinpath(path, archivename)
+        if isfile(archivepath)
+            if checksha(ver, archivepath)
+                return ver
+            else
+                error("Archive found but didn't match SHA for $(ver.algname).")
+            end
+        end
+    end
+    return nothing 
+end
+
 # Parse some basic command-line arguments
 const verbose = "--verbose" in ARGS
 const prefix = Prefix(get([a for a in ARGS if a != "--verbose"], 1, joinpath(@__DIR__, "usr")))
 
-const hsl_ma57_version = "5.2.0"
-const hsl_ma57_sha256 = "aedc5a3e22a7b86779efccaa89a7c82b6949768dbab35fceb85a347e326cf584"
+##############################
+# MA57
+##############################
+const hsl_ma57_versions = [
+    HSLVersion("hsl_ma57", "5.2.0", "aedc5a3e22a7b86779efccaa89a7c82b6949768dbab35fceb85a347e326cf584", ".tar.gz"),
+]
 
 const hsl_ma57_path = haskey(ENV, "HSL_MA57_PATH") ? ENV["HSL_MA57_PATH"] : joinpath(@__DIR__, "downloads")
-const hsl_ma57_archive = joinpath(hsl_ma57_path, "hsl_ma57-$hsl_ma57_version.tar.gz")
+hsl_ma57_version = findversion(hsl_ma57_versions, hsl_ma57_path)
+const hsl_ma57_archive = isnothing(hsl_ma57_version) ? "" : joinpath(hsl_ma57_path, getname(hsl_ma57_version))
 const hsl_ma57_patch = joinpath(hsl_ma57_path, "get_factors.patch")
 
-const hsl_ma97_version = "2.6.0"
-const hsl_ma97_sha256 = "be5fe822674be93e3d2e1a7d7ed6c5ad831b91cf8ca5150beb473f67af5fcb66"
-
+##############################
+# MA97
+##############################
+const hsl_ma97_verions = [
+    HSLVersion("hsl_ma97", "2.6.0", "be5fe822674be93e3d2e1a7d7ed6c5ad831b91cf8ca5150beb473f67af5fcb66", ".tar.gz"),
+    HSLVersion("hsl_ma97", "2.7.0", "ac3a081d3a28e9ecb8871ce769f4ced2a5ffa5a9c36defbd2c844ae3493ccb37", ".tar.gz"),
+    HSLVersion("hsl_ma97", "2.7.0", "8221b607d96554d7a57cc60483c7305ef43a8785dc4171ac2e8da087900a1100", ".zip")
```

Comment:
What happens if the user drops two or more versions under `ENV["HSL_MA97_PATH"]` or `downloads`? It will just pick the first one, right? Shouldn't we list the most recent on top then?

Reply:
Yep, they should be ordered in priority. I've redorded them so that the latest version is now first.